packages/react/src/CTAForm/CTAForm.test.tsx

```diff
@@ -0,0 +1,167 @@
+    const mockTestId = 'test'
+    const expectedClass = 'test'
+
+    const {getByTestId} = render(
+      <CTAForm
+        data-testid={mockTestId}
+        className={expectedClass}
+        action={expectedAction}
+        method={expectedMethod}
+        target={expectedTarget}
+      >
+        <CTAForm.Action>submit</CTAForm.Action>
+      </CTAForm>
+    )
+    const ctaFormEl = getByTestId(mockTestId)
+    expect(ctaFormEl).toHaveAttribute('action', expectedAction)
+    expect(ctaFormEl).toHaveAttribute('method', expectedMethod)
+    expect(ctaFormEl).toHaveAttribute('target', expectedTarget)
+    expect(ctaFormEl.classList).toContain(expectedClass)
+  })
+
+  it('submits the form correctly', () => {
+    const expectedAction = 'https://www.github.com'
+    const expectedMethod = 'POST'
+    const expectedTarget = '_blank'
+    const mockTestId = 'test'
+    const expectedClass = 'test'
+    const mockSubmit = jest.fn(e => e.preventDefault())
+    const submitText = 'Submit'
+
+    const {getByTestId} = render(
+      <CTAForm
+        className={expectedClass}
+        action={expectedAction}
+        method={expectedMethod}
+        target={expectedTarget}
+        onSubmit={mockSubmit}
+      >
+        <CTAForm.Action data-testid={mockTestId}>{submitText}</CTAForm.Action>
+      </CTAForm>
+    )
+    fireEvent(
+      getByTestId(mockTestId),
+      new MouseEvent('click', {
+        bubbles: true,
+        cancelable: true
+      })
+    )
+    expect(mockSubmit).toHaveBeenCalledTimes(1)
+  })
+})
+
+describe('CTAForm.Input', () => {
+  afterEach(cleanup)
+
+  it('no a11y violations', async () => {
+    const {container} = render(<CTAForm.Input></CTAForm.Input>)
+    const results = await axe(container)
+
+    expect(results).toHaveNoViolations()
+  })
+
+  it('allows and passes expected attributes to the CTAForm.Input component', async () => {
+    const mockTestId = 'test'
+    const expectedClass = 'test'
+
+    const {getByTestId} = render(<CTAForm.Input data-testid={mockTestId} className={expectedClass}></CTAForm.Input>)
+    const ctaInputEl = getByTestId(mockTestId)
+    expect(ctaInputEl.classList).toContain(expectedClass)
+  })
+})
+
+describe('CTAForm.Confirm', () => {
+  afterEach(cleanup)
+
+  it('no a11y violations', async () => {
+    const {container} = render(<CTAForm.Confirm></CTAForm.Confirm>)
+    const results = await axe(container)
+
+    expect(results).toHaveNoViolations()
+  })
+
+  it('allows and passes expected attributes to the CTAForm.Confirm component', async () => {
+    const mockTestId = 'test'
+    const expectedClass = 'test'
+
+    const {getByTestId} = render(<CTAForm.Confirm data-testid={mockTestId} className={expectedClass}></CTAForm.Confirm>)
+    const ctaConfirmEl = getByTestId(mockTestId)
+    expect(ctaConfirmEl.classList).toContain(expectedClass)
+  })
+})
+
+describe('CTAForm.Action', () => {
+  afterEach(cleanup)
+
+  it('no a11y violations', async () => {
```

Comment:
Do we need to test each child individually? Wonder if these tests can be rolled into the main suite at the top?

Reply:
It depends on how logging works. Would it roll up all of the accessibility issues into the main test block? Would it be easier for a user to view them this way or individually by component piece?

Reply:
> It depends on how logging works. 

sorry, which logging? 

I was referring to the approach of using different describe blocks for testing each child independently. I haven't seen that done elsewhere, so wondering what the motivation was for doing it here. I don't think it necessarily has a user impact 🤔, but it could IMO make it harder to maintain those test fails given how much longer they are. If we scale it out to other components also, it could make our suites slower to complete, so wondering what the benefits were.

Another reason for asking, is that the child components aren't meant to be used in isolation, but rather part of the wider CTAForm. This is usually important context for axe runners.

Reply:
I would be happy to consolidate for consistency!